From 8740d28bfeb63dd021b390a8c7a7c7a6f25c362c Mon Sep 17 00:00:00 2001
From: Nan Wu <wnan@google.com>
Date: Wed, 10 Sep 2025 21:23:55 +0000
Subject: [PATCH] RESTRICT AUTOMERGE: Backport Cut max duration for default
 transition to 1500

3 seconds is way too long, especially for tap jacking exploitation during animation. Cut it down to 1500 for a more tolerable experience.

Bug: 266433089
Test: manual
Flag: EXEMPT bug fix
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:1e9412266ede59c046b83ad3a3fbfcaba94a1787
Merged-In: I3b33e47743fc2dee93cb3cb6363626698868beae
Change-Id: I3b33e47743fc2dee93cb3cb6363626698868beae
---
 .../android/wm/shell/transition/DefaultTransitionHandler.java   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java b/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
index 9154226b7b22..a4f77f6a68ba 100644
--- a/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
+++ b/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
@@ -114,7 +114,7 @@ import java.util.function.Consumer;
 
 /** The default handler that handles anything not already handled. */
 public class DefaultTransitionHandler implements Transitions.TransitionHandler {
-    private static final int MAX_ANIMATION_DURATION = 3000;
+    private static final int MAX_ANIMATION_DURATION = 1500;
 
     /**
      * Restrict ability of activities overriding transition animation in a way such that
-- 
2.52.0.rc1.455.g30608eb744-goog

